src/libs/actions/PersistedRequests.ts

```diff
@@ -17,15 +17,27 @@ function clear() {
     return Onyx.set(ONYXKEYS.PERSISTED_REQUESTS, []);
 }
 
-function save(requestsToPersist: Request[]) {
-    let requests: Request[] = [];
-    if (persistedRequests.length) {
-        requests = persistedRequests.concat(requestsToPersist);
+function createUpdatedRequest(oldRequest: Request, newRequest: Request): Request {
+    // Merge the requests together, but concat Onyx update arrays together
+    return {
```

Comment:
This will lose any properties that are present in `oldRequest` but not `newRequest`. Spreads also do a shallow copy instead of a deep copy. Let's please use `lodashMerge` instead and look for a workaround for the TS `any` issue if possible. I'd rather use `tsignore` than implement the wrong behavior or implement a custom deep merge.

Reply:
I've refactored this method. Now it uses `lodashMergeWith` function. Do we also want to add an additional condition in this function to prevent from adding duplicated elements to PersistedRequests array?

Reply:
No, I think this is good the way it is now.